src/components/safe-apps/SafeAppLandingPage/AppActions.tsx

```diff
@@ -1,38 +1,59 @@
 import { Box, Button, MenuItem, Select, Typography, Grid, FormControl, InputLabel } from '@mui/material'
-import type { ChainInfo } from '@gnosis.pm/safe-react-gateway-sdk'
-import { useState } from 'react'
+import type { ChainInfo, SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useEffect, useMemo, useState } from 'react'
 import type { ConnectedWallet } from '@/hooks/wallets/useOnboard'
 import { useAppSelector } from '@/store'
-import { selectAddressBookByChain } from '@/store/addressBookSlice'
+import { selectAllAddressBooks } from '@/store/addressBookSlice'
+import { selectChains } from '@/store/chainsSlice'
 import useLastSafe from '@/hooks/useLastSafe'
 import { parsePrefixedAddress } from '@/utils/addresses'
 import SafeIcon from '@/components/common/SafeIcon'
 import EthHashInfo from '@/components/common/EthHashInfo'
 import { AppRoutes } from '@/config/routes'
+import useOwnedSafes from '@/hooks/useOwnedSafes'
 import { CTA_BUTTON_WIDTH, CTA_HEIGHT } from '@/components/safe-apps/SafeAppLandingPage/constants'
 
 type Props = {
   appUrl: string
   wallet: ConnectedWallet | null
   onConnectWallet: () => Promise<void>
-  safes: string[]
   chain: ChainInfo
+  app: SafeAppData
 }
 
-const AppActions = ({ wallet, onConnectWallet, safes, chain, appUrl }: Props): React.ReactElement => {
+type CompatibleSafesType = { address: string; chainId: string; shortName?: string }
+
+const AppActions = ({ wallet, onConnectWallet, chain, appUrl, app }: Props): React.ReactElement => {
   const lastUsedSafe = useLastSafe()
-  const lastUsedSafeAddress =
-    lastUsedSafe && safes.includes(lastUsedSafe) ? parsePrefixedAddress(lastUsedSafe).address : ''
-  const [safeToUse, setSafeToUse] = useState<string | undefined>(lastUsedSafeAddress || undefined)
-  const addressBook = useAppSelector((state) => selectAddressBookByChain(state, chain.chainId))
+  const ownedSafes = useOwnedSafes()
+  const addressBook = useAppSelector(selectAllAddressBooks)
+  const chains = useAppSelector(selectChains)
+  const compatibleChains = app.chainIds
+
+  const compatibleSafes = useMemo(
+    () => getCompatibleSafes(ownedSafes, compatibleChains, chains.data),
+    [ownedSafes, compatibleChains, chains.data],
+  )
+
+  const [safeToUse, setSafeToUse] = useState<CompatibleSafesType>()
+
+  useEffect(() => {
+    const selectedSafe = getDefaultSafe(compatibleSafes, chain.chainId, lastUsedSafe)
```

Comment:
if the function is named `getDefaultSafe`, would it be better to name the variable `defaultSafe`?

Reply:
done!